common/src/main/java/com/dfsek/terra/api/util/world/PaletteUtil.java

```diff
@@ -2,16 +2,19 @@
 
 import com.dfsek.terra.api.math.MathUtil;
 import com.dfsek.terra.api.world.palette.Palette;
-import com.dfsek.terra.api.world.palette.holder.PaletteHolder;
+import com.dfsek.terra.api.world.palette.slant.SlantHolder;
 import com.dfsek.terra.config.templates.BiomeTemplate;
 import com.dfsek.terra.world.generation.math.samplers.Sampler;
 
 public final class PaletteUtil {
     public static Palette getPalette(int x, int y, int z, BiomeTemplate c, Sampler sampler) {
-        PaletteHolder slant = c.getSlantPalette();
-        if(slant != null && MathUtil.derivative(sampler, x, y, z) > c.getSlantThreshold()) {
-            return slant.getPalette(y);
+
+        SlantHolder slant = c.getSlant();
+        double slope = MathUtil.derivative(sampler, x, y, z);
```

Comment:
Performance - put this inside the if statement conditional. We don't want to calculate the derivative if we dont need it.